src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,36 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List getAll(Class<?> type) {
+        if (type.equals(Bulldozer.class)) {
+            return new BulldozerProducer().get();
+        } else if (type.equals(Excavator.class)) {
+            return new ExcavatorProducer().get();
+        } else if (type.equals(Truck.class)) {
+            return new TruckProducer().get();
```

Comment:
Introduce your `Producer`s as field variables and initialize in constructor.
Remove `else` keywords, you have returns from `if` blocks